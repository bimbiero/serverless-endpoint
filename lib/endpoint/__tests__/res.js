'use strict';

/* global describe jest test expect */
jest.dontMock('../res');

const Res = require('../res');

describe('endpoint.res.js', () => {
  const message = {
    done: jest.fn(),
    error: jest.fn()
  };

  test('creates send and error from message', () => {
    const req = new Res(message);
    expect(req.send).toBeDefined();
    expect(req.error).toBeDefined();
  });

  test('creates send from done function', () => {
    const req = new Res(message);
    const body = { message: 'body' };
    req.send(200, body);
    expect(message.done).toBeCalled();
  });

  test('send stringify body', () => {
    const req = new Res(message);
    const body = { message: 'body' };
    req.send(200, body);
    expect(message.done).toBeCalledWith({ statusCode: 200, body: JSON.stringify(body) });
  });

  test('error calls message.error', () => {
    const error = console.error;
    console.error = jest.fn();
    const req = new Res(message);
    req.error(new Error('something went wrong'));
    expect(message.error).toBeCalled();
    expect(console.error).toMatchSnapshot();
    console.error = error;
  });

});
