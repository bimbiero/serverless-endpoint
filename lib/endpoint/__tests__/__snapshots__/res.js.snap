// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`endpoint.res.js error calls message.error 1`] = `[Function]`;
